### PR TITLE
Fix default config reading import

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
The original import won't work in stored procedure, so we need to import it inside the create function

### CODE REVIEW

File: src/snowflake/snowpark/session.py [hunk 1]
```diff
@@ -21,7 +21,6 @@
 import pkg_resources
 
 from snowflake.connector import ProgrammingError, SnowflakeConnection
-from snowflake.connector.config_manager import _get_default_connection_params
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/session.py:24]: We probably can do `is_in_stored_procedure()` check for the import? And for safety, also add the same check in the if clause before calling the function.

**Review by sfc-gh-sfan — APPROVED** [review]
We might want to add this function to sp connector even if it is just a no-op to reduce the stored proc check we have in code. Can you create a ticket to track that?

**Review by sfc-gh-sfan — CHANGES_REQUESTED** [review]
Sorry does not meant to approve yet

File: src/snowflake/snowpark/session.py (hunk 1 above)
Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/session.py:24]: done

**Comment by sfc-gh-jdu** [comment]
> We might want to add this function to sp connector even if it is just a no-op to reduce the stored proc check we have in code. Can you create a ticket to track that?

sure, created https://snowflakecomputing.atlassian.net/browse/SNOW-911000

**Review by sfc-gh-sfan — APPROVED** [review]

**Comment by sfc-gh-aalam** [comment]
```
2023-09-08 11:16:32     from snowflake.snowpark.session import Session
2023-09-08 11:16:32   File "/home/jenkins/workspace/AnacondaPackageBuilder-x86/conda-bld/snowflake-snowpark-python_1694196935262/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.8/site-packages/snowflake/snowpark/session.py", line 154, in <module>
2023-09-08 11:16:32     from snowflake.connector.config_manager import _get_default_connection_params
2023-09-08 11:16:32 ImportError: cannot import name '_get_default_connection_params' from 'snowflake.connector.config_manager' (/home/jenkins/workspace/AnacondaPackageBuilder-x86/conda-bld/snowflake-snowpark-python_1694196935262/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.8/site-packages/snowflake/connector/config_manager.py)
```

This is still breaking https://ci-dev-142.int.snowflakecomputing.com/job/AnacondaPackageBuilder-x86/

@sfc-gh-jdu can you look into this and verify if our precommit test is happy?

**Pull request merged**